### PR TITLE
OPENEUROPA-1993: ECL2 Update gallery.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Update gallery component to ECL2

### Change log

- Added:
- Changed: Update gallery component to ECL2
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: js/gallery.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/**
+ * @file
+ * ECL gallery behavior.
+ */
+(function (ECL, Drupal) {
+  Drupal.behaviors.eclGallery = {
+    attach: function attach() {
+      var galleryElements = document.querySelectorAll("[data-ecl-gallery]");
```
Comment (voidtek) [inline js/gallery.js]: To be consistent
```suggestion
      var elements = document.querySelectorAll('[data-ecl-accordion]');
      elements.forEach(function(element) {
        (new ECL.Accordion(element)).init();
      });
```

File: templates/patterns/gallery/gallery.ui_patterns.yml [hunk 2]
```diff
@@ -7,34 +7,119 @@ gallery:
       label: "Items"
       description: "List of GalleryItemValueObject objects."
       preview:
-        - thumbnail:
```
Comment (voidtek) [inline templates/patterns/gallery/gallery.ui_patterns.yml]: We should keep the same structure of the old pattern.

**Review by voidtek — CHANGES_REQUESTED** [review]

File: templates/patterns/gallery/gallery.ui_patterns.yml (hunk 2 above)
Comment (imanoleguskiza) [reply templates/patterns/gallery/gallery.ui_patterns.yml]: We are keeping the same structure though, we still have a list of GalleryItemValueObjects. The objects themselves have changed, which I think its ok?

**Comment by imanoleguskiza** [comment]
Outdated PR

**Pull request closed without merging**